FirebaseDatabase/Sources/Core/FRepo.m

```diff
@@ -533,7 +533,7 @@ - (void)getData:(FIRDatabaseQuery *)query
         }];
         return;
     }
-    [self.persistenceManager setQueryActive:querySpec];
+    NSNumber *tag = [self.serverSyncTree registerQuery:[query querySpec]];
     [self.connection
```

Comment:
Hi @IanWyszynski ,
I just had a small question about the deleted line 536:

```
[self.persistenceManager setQueryActive:querySpec];
```

There is still a 'matching':
```
[self.persistenceManager setQueryInactive:querySpec];
```
On the new line 595.

I am a bit confused by the previous code since there is a case of an 'early return' (old line number 564) where the query would NOT be set inactive again. This could be a bug.

All in all: I don't know what the correct behavior is, but I do think that there was already something fishy about the setQueryActive and setQueryInactive and the early return, and after this change there seems to be even more of an asymmetry here. :-)